### PR TITLE
[FT][LLM tests]Add FT integration tests for gpt2, opt and bloom models

### PR DESCRIPTION
**Pull request by rohithkrn**
## Description ##

Add FT integration tests for gpt2, opt and bloom models

### CODE REVIEW

File: tests/integration/llm/prepare.py [hunk 1]
```diff
@@ -81,7 +81,19 @@
...
         "option.model_id": "t5-small",
         "option.tensor_parallel_degree": 4,
-    }
+    },
+    "gpt2-xl": {
+        "option.model_id": "gpt2-xl",
+        "option.tensor_parallel_degree": 1,
+    },
+    "facebook/opt-13b": {
+        "option.model_id": "facebook/opt-13b",
```
Comment (lanking520) [inline tests/integration/llm/prepare.py:90]: This model is in our S3 bucket, using s3 bucket downloading will speed up

File: tests/integration/llm/prepare.py [hunk 2]
```diff
@@ -81,7 +81,19 @@
...
+        "option.tensor_parallel_degree": 4,
+    },
+    "bigscience/bloomz-3b": {
+        "option.model_id": "bigscience/bloomz-3b",
```
Comment (lanking520) [inline tests/integration/llm/prepare.py:94]: bloom 7b is in the bucket

**Comment by rohithkrn** [comment]
the test job failed for gpt2-xl with error: `RuntimeError: context has already been set`. Let's wait for the fix to merge in.
https://github.com/deepjavalibrary/djl-serving/actions/runs/4148142635/jobs/7175851331

File: tests/integration/llm/prepare.py (hunk 1 above)
Comment (rohithkrn) [reply tests/integration/llm/prepare.py]: updated

File: tests/integration/llm/prepare.py (hunk 2 above)
Comment (rohithkrn) [reply tests/integration/llm/prepare.py]: updated

**Comment by rohithkrn** [comment]
Passing test logs - https://github.com/deepjavalibrary/djl-serving/actions/runs/4167577542/jobs/7213363647.
It's taking several tries to connect djl-serving for opt and bloom models.

File: tests/integration/llm/client.py [hunk 3]
```diff
@@ -76,7 +76,19 @@
...
+        "batch_size": [1, 2],
+        "max_memory_per_gpu": 7.0
+    },
+    "facebook/opt-13b": {
+        "batch_size": [1, 2],
+        "max_memory_per_gpu": 15.0
+    },
+    "bigscience/bloom-7b1": {
+        "batch_size": [1, 2],
+        "max_memory_per_gpu": 17.0
```
Comment (lanking520) [inline tests/integration/llm/client.py:90]: please use fp16 for testing. Also the artifact saving in this as well. Ideally, we are trying to keep a GPU memory usage under 50% (12GB).

File: tests/integration/llm/client.py [hunk 4]
```diff
@@ -76,7 +76,19 @@
...
         "max_memory_per_gpu": 2
-    }
+    },
+    "gpt2-xl": {
+        "batch_size": [1, 2],
+        "max_memory_per_gpu": 7.0
+    },
+    "facebook/opt-13b": {
+        "batch_size": [1, 2],
+        "max_memory_per_gpu": 15.0
```
Comment (lanking520) [inline tests/integration/llm/client.py:86]: same in here. the model is saved in fp16, not making sense to test in fp32

File: tests/integration/llm/client.py (hunk 3 above)
Comment (rohithkrn) [reply tests/integration/llm/client.py:90]: makes sense, these tests are written when only fp32 support was tested/present. Before updating these tests, we need to update the wheel with latest changes I guess.

File: tests/integration/llm/client.py (hunk 4 above)
Comment (rohithkrn) [reply tests/integration/llm/client.py]: updated.

File: tests/integration/llm/client.py (hunk 3 above)
Comment (rohithkrn) [reply tests/integration/llm/client.py]: updated.

**Comment by rohithkrn** [comment]
Passing test logs - https://github.com/deepjavalibrary/djl-serving/actions/runs/4179684457/jobs/7239925371

**Review by lanking520 — APPROVED** [review]

**Pull request merged**